test/tests_for_integration/api_to_db_test.py

```diff
@@ -84,12 +104,65 @@ def auth_url(config, mongo_client):
     _clean_auth_db(mongo_client)
 
 
-# TODO start the ee2 service
-# TODO wipe the ee2 database between every test
+# side effect: modifies the config
+def _create_config_file(full_config, auth_url):
+    ee2c = full_config[EE2_CONFIG_SECTION]
+    ee2c["auth-service-url"] = auth_url + "/api/legacy/KBase/Sessions/Login"
+    ee2c["auth-service-url-v2"] = auth_url + "/api/v2/token"
+    ee2c["auth-url"] = auth_url
+    ee2c["auth-service-url-allow-insecure"] = "true"
+
+    deploy = tempfile.mkstemp(".cfg", "deploy-", dir=TEMP_DIR, text=True)
+    os.close(deploy[0])
+
+    with open(deploy[1], "w") as handle:
+        full_config.write(handle)
+
+    return deploy[1]
+
+
+def _clear_ee2_db(mc: pymongo.MongoClient, config: Dict[str, str]):
+    ee2 = mc[config["mongo-database"]]
+    for name in ee2.list_collection_names():
+        if not name.startswith("system."):
+            # don't drop collection since that drops indexes
+            ee2.get_collection(name).delete_many({})
+
+
+@fixture(scope="module")
+def service(full_config, auth_url, mongo_client, config):
+    cfgpath = _create_config_file(full_config, auth_url)
+    _clear_ee2_db(mongo_client, config)
+
+    # from this point on, calling the get_*_test_config methods will get the temp config file
+    os.environ[KB_DEPLOY_ENV] = cfgpath
+    portint = find_free_port()
+    Thread(
+        target=execution_engine2Server.start_server,
+        kwargs={"port": portint},
+        daemon=True,
+    ).start()
+    time.sleep(0.05)
+    port = str(portint)
+    print("running ee2 service at localhost:" + port)
+    yield port
+
+    # shutdown the server
+    # SampleServiceServer.stop_server()  <-- this causes an error. the start & stop methods are
+    # bugged. _proc is only set if newprocess=True
```

Comment:
Would this make more sense if I looked at the rest of the file? It seems a bit of a non-sequitur as-is -- what is `_proc`? It could do with being rewritten to be clearer.

Reply:
Rewrote